lib/koans/15_protocols.ex

```diff
@@ -0,0 +1,39 @@
+defmodule Protocols do
+  use Koans
+
+  defprotocol School do
+    def enrol(name)
+  end
+
+  defimpl School, for: SecondarySchool do
+    def enrol(name) do
+      "#{name} enrolled at secondary school"
+    end
+  end
+
+  defimpl School, for: DanceSchool do
+    def enrol(name) do
+      "#{name} enrolled for dance"
+    end
+  end
+
+  defmodule EveningClass do end
+
+  @schools %{
+    :secondary => School.SecondarySchool,
+    :dance => School.DanceSchool,
+    :evening => EveningClass
+  }
+
+  koan "Commonality is the secret" do
+    school = Map.get(@schools, :dance)
```

Comment:
Ah hah! I realized what the problem is. You don't utilize a protocol by calling functions on the module, you do it by calling the function on the protocol.

So instead of saying `School.DanceSchool.enrol("Jay")`, you would say `School.enrol(student)` where student would be a value with a protocol defined for it. Does that make sense?

Take a closer look at http://elixir-lang.org/getting-started/protocols.html. The protocol is exercised by calling Blank.blank?/1 with different types of arguments.

```
iex> Blank.blank?(0)
false
iex> Blank.blank?([])
true
iex> Blank.blank?([1, 2, 3])
false
iex> Blank.blank?("hello")
** (Protocol.UndefinedError) protocol Blank not implemented for "hello"
iex(3)> Blank.String.blank?("no")
** (UndefinedFunctionError) undefined function Blank.String.blank?/1 (module Blank.String is not available)
    Blank.String.blank?("no")
```

Calling methods directly on the specific type, e.g. Blank.Atom.blank?/1, defeats the purpose.